src/components/ProductRegistrationForm/Form.tsx

```diff
@@ -0,0 +1,272 @@
+import React, { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import { useForm } from 'react-hook-form'
+import TextField from '../InputField/TextField'
+import useTranslation from 'next-translate/useTranslation'
+import { ErrorsType, ImageListType } from 'react-images-uploading'
+import ProductUpload from '../ProductUpload'
+import Dropdown, { Option } from 'react-dropdown'
+import TextArea from '../Textarea'
+
+interface FormInput {
+  readonly name: string
+  readonly price: string
+  readonly color: string
+  readonly size: string
+  readonly description: string
+}
+
+const initialInputValue: FormInput = {
+  name: '',
+  price: '',
+  color: '',
+  size: '',
+  description: ''
+}
+
+const INPUT_TEXT_VALIDATION_INDEX: number = 2
+const INPUT_NUMBER_VALIDATION_INDEX: number = 1
+const NUMBER_ONLY_PATTERN_VALUE = new RegExp(/^(?:[1-9]\d*|0)$/)
```

Comment:
This one should also handle with decimal number.
```suggestion
const NUMBER_ONLY_PATTERN_VALUE = new RegExp(/^[1-9]\d*(\.\d+)?$/)
```

Reply:
Please see this [comment](https://github.com/Cobda/cobda-web/pull/55#discussion_r619603521)